src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -215,15 +215,19 @@ void OwnershipBasedObjectDirectory::SubscriptionCallback(
     }
   }
 
-  auto worker_it = worker_rpc_clients_.find(worker_id);
-  rpc::GetObjectLocationsOwnerRequest request;
-  request.set_intended_worker_id(worker_id.Binary());
-  request.set_object_id(object_id.Binary());
-  request.set_last_version(reply.current_version());
-  worker_it->second->GetObjectLocationsOwner(
-      request,
-      std::bind(&OwnershipBasedObjectDirectory::SubscriptionCallback, this, object_id,
-                worker_id, std::placeholders::_1, std::placeholders::_2));
+  // Only send the next long-polling RPC if the last one was successful.
```

Comment:
Isn't it possible that just other simple failures can cause issues? Why doesn't the owner just sends a cancel subscription request to all subscribers? Or can we use a specific status to decide if we'd like to cancel subscription? (e.g., instead of status.ok(), we can do something like !status.NotFound())

Reply:
Ignore this. I thought this PR was for something else.

Reply:
So, you translated !status.ok() as an owner failure right? Can you write more detailed comment here?

Reply:
> Isn't it possible that just other simple failures can cause issues?

We've already decided that any failed RPCs to the owner are going to be considered equivalent to the object being freed. Are you wanting to revisit that?

>Why doesn't the owner just sends a cancel subscription request to all subscribers?

That's not a resilient protocol, since the owner could have died, which would be equivalent to the object being freed. And subscription cancellation is only supposed to be down by the subscribers, and there's already logic to do that.

>Or can we use a specific status to decide if we'd like to cancel subscription? (e.g., instead of status.ok(), we can do something like !status.NotFound())

We don't have visibility into that status, [they all come back as `IOError` statuses](https://github.com/ray-project/ray/blob/a0f73cf3f72d02b5f47f4a189052171a3fdd85c3/src/ray/common/grpc_util.h#L78-L89).

Reply:
oh please ignore the first comment haha

Reply:
>So, you translated !status.ok() as an owner failure right? Can you write more detailed comment here?

This can also happen if the object was freed, since that will come back as an gRPC error, and the object being freed does also encapsulate the owner failing (we consider the object to have been freed if the owner fails). Do you think that we need to explicitly call out an owner failure here, or is the comment about the object being considered freed enough since it covers both cases?

Reply:
Ah, that makes sense. I think the current comment is fine.